sphinx_needs/builder.py

```diff
@@ -157,3 +158,66 @@ def build_needumls_pumls(app: Sphinx, _exception: Exception) -> None:
         needs_builder.set_environment(env)
 
     needs_builder.finish()
+
+
+class NeedsLookUpTableBuilder(Builder):
+    name = "needs_lut"
+    format = "needs"
+    file_suffix = ".txt"
+    links_suffix = None
+
+    def write_doc(self, docname: str, doctree: nodes.document) -> None:
+        pass
+
+    def finish(self) -> None:
+        env = unwrap(self.env)
+        needs = env.needs_all_needs.values()
+        needs_dict = {}
+        for need in needs:
+            if need["is_external"]:
+                needs_dict[need["id"]] = need["external_url"]
+            else:
+                needs_dict[need["id"]] = need["docname"]
+
+        try:
+            fname = os.path.join(self.outdir, "needs_lut.json")
+            with open(fname, "w", encoding="utf-8") as f:
+                json.dump(needs_dict, f, indent=4)
+        except Exception as e:
```

Comment:
Can we add this part (dict creation + json.dump) also to NeedsList as  ``write_json_lut``?
It's easier to maintain if all storage functions are defined in one place.

Reply:
I think we should stay with `add_need` (instead of `add_lut_need`) and transform the needed data in a special writer function `write_json_lut`.